src/js/initializers/pluginManager.ts

```diff
@@ -56,4 +37,56 @@ export default class PluginManager {
   async deactivate(): Promise<void> {
     await Promise.all(Object.values(this.plugins).map((p) => p.deactivate()))
   }
+
+  private async add(plugin: Plugin) {
+    if (this.plugins[plugin.name]) {
+      console.error(
+        new Error(`Duplicate or name collision for '${plugin.name}' plugin`)
+      )
+    } else {
+      this.plugins[plugin.name] = plugin
+    }
+  }
+}
+
+class Plugin {
+  name: string
+  entryPoint: string
+  functions: PluginFunctions
+
+  constructor(public folderPath: string) {
+    this.name = path.basename(folderPath)
+    this.entryPoint = path.join(folderPath, this.indexFile())
+  }
+
+  async validate() {
+    const d = lib.file(this.folderPath)
+    if (!(await d.isDirectory())) {
+      return "Plugin must be contained in a directory."
+    }
+    if (!lib.file(this.entryPoint).existsSync()) {
+      return "Plugin directory must contain an index file as the entry point."
+    }
+    return null
+  }
+
+  load() {
+    this.functions = require(this.entryPoint)
+  }
+
+  activate(api: BrimApi) {
+    this.functions.activate(api)
+  }
+
+  deactivate() {
+    if ("deactivate" in this.functions) {
+      this.functions.deactivate()
+    }
+  }
+
+  private indexFile() {
+    if (process.env.BRIM_ITEST === "true") return "index.js"
+    if (process.env.NODE_ENV === "test") return "index.ts"
+    return "index.js"
```

Comment:
This is the bug. 
* In integration tests, we want index.js
* In unit tests we want index.ts 
* In production/dev we want index.js.